### ISSUE TITLE
suggested adjustments to template example

### ISSUE THREAD

**Issue by kewalaka** [Azure/terraform-azurerm-avm-template#38]
A couple changes are suggested to the template example

1) add an azurerm features block, I prefer the practice of skip provider registrations and registering ones that are required explicitly as part of vending, so something like;

```terraform
provider "azurerm" {
  skip_provider_registration = true
  features {}
}
```

2) make the minimum version of Terraform 1.3.0, because, it's extremely likely that "optional" is going to be used.

**Comment by matt-FFFFFF** [comment]
Hi,

Setting provider blocks is not something we want to do inside a module, preferring to leave this to the consumer.

I take your point on the min TF version though. Will update.

**Comment by matt-FFFFFF** [comment]
> Provider configurations belong in the root module of a Terraform configuration

ref: https://developer.hashicorp.com/terraform/language/providers/configuration#provider-configuration-1

**Comment by matt-FFFFFF** [comment]
updated min tf version in #41

**Comment by matt-FFFFFF** [comment]
Apologies you were referring to the example - will fix

**Comment by matt-FFFFFF** [comment]
fixed in #42
